src/thread.cpp

```diff
@@ -89,7 +89,7 @@ namespace libtorrent
 		ts.tv_sec = tv.tv_sec + total_seconds(rel_time) + microseconds / 1000000;
 
 		// wow, this is quite a hack
-		pthread_cond_timedwait(&m_cond, reinterpret_cast<::pthread_mutex_t*>(&l.mutex()), &ts);
```

Comment:
this is quite ugly. I should probably switch to boost.thread (assuming it's conformant to/wraps std::thread these days)